### PR TITLE
Add: Support for sampling events

### PR DESCRIPTION
**Pull request by keyur9**
Add: Support for sampling events
Refactor: Fix to capture request body
Refactor: README.md

### CODE REVIEW

File: MoesifWorker.js [hunk 1]
```diff
@@ -448,6 +495,13 @@ async function logRequest(event) {
     return;
   }
 
+  var randomNumber = Math.random() * 100;
+  // Fetch app Config first time or if not fetched within last 5 mins
+  if (!isAppConfigFetched || lastUpdatedAppConfigTime == null || new Date().getTime() > lastUpdatedAppConfigTime + fetchAppConfigTimeDeltaInMins ) {
+    lastUpdatedAppConfigTime = new Date().getTime();
+    event.waitUntil(fetchAppConfig());
```
Comment (dkm199) [inline MoesifWorker.js:503]: does event need to wait for config to be fetched? Or is there a way to do this asycnhronously? like in await , and not wait for await?

Comment (keyur9) [reply MoesifWorker.js:503]: It doesn't block the response - https://developers.cloudflare.com/workers/learning/fetch-event-lifecycle#waituntil

File: MoesifWorker.js [hunk 2]
```diff
@@ -471,25 +528,49 @@ async function logRequest(event) {
...
+  samplingPercentage = getSamplingPercentage(userId, companyId);
+  if (randomNumber > samplingPercentage) {
+    moesifLog('Skip sending event to Moesif due to sampling percentage');
     return response;
+  } else {
+    const after = new Date();
+    const txId = request.headers.get(TRANSACTION_ID_HEADER) || uuid4();
+  
+    event.waitUntil(
+      tryTrackRequest(
```
Comment (dkm199) [inline MoesifWorker.js:556]: can you add `weight` to the event, its needed for various purposes in backend/frontend

**Review by dkm199 — APPROVED** [review]
looks good, need to add weight

File: MoesifWorker.js (hunk 2 above)
Comment (keyur9) [reply MoesifWorker.js:556]: added weight to the event model

File: MoesifWorker.js [hunk 3]
```diff
@@ -320,7 +320,8 @@ async function makeMoesifEvent(request, response, before, after, txId, requestBo
       status: response.status,
       headers: headersToObject(response.headers),
     },
-    direction: response.isEmpty ? 'Incoming' : 'Outgoing' 
+    direction: response.isEmpty ? 'Incoming' : 'Outgoing',
+    weight: 1
```
Comment (dkm199) [inline MoesifWorker.js]: hmm, isn't weight a fcn of sampling percentage, when sampling is enabled?

**Review by dkm199 — APPROVED** [review]

**Pull request merged**